src/taipy/core/config/data_node_config.py

```diff
@@ -310,222 +351,232 @@ def _configure(
                 _storage_type_ value set in the default data node configuration
                 (see `(Config.)configure_default_data_node()^`)), _"pickle"_, _"csv"_, _"excel"_, _"sql_table"_,
                 _"sql"_, _"json"_, _"parquet"_, _"mongo_collection"_, _"in_memory"_, or _"generic"_.
-            scope (Scope^): The scope of the data node configuration. The default value is
+            scope (Optional[Scope^]): The scope of the data node configuration. The default value is
                 `Scope.SCENARIO` (or the one specified in
                 `(Config.)configure_default_data_node()^`).
             **properties (Dict[str, Any]): A keyworded variable length list of additional
                 arguments.
         Returns:
             `DataNodeConfig^`: The new data node configuration.
         """
-        section = DataNodeConfig(id, storage_type, scope, **properties)
-        Config._register(section)
-        return Config.sections[DataNodeConfig.name][id]
+        configuration_map: Dict[str, Callable] = {
+            cls._STORAGE_TYPE_VALUE_PICKLE: cls._configure_pickle,
+            cls._STORAGE_TYPE_VALUE_SQL_TABLE: cls._configure_sql_table,
+            cls._STORAGE_TYPE_VALUE_SQL: cls._configure_sql,
+            cls._STORAGE_TYPE_VALUE_MONGO_COLLECTION: cls._configure_mongo_collection,
+            cls._STORAGE_TYPE_VALUE_CSV: cls._configure_csv,
+            cls._STORAGE_TYPE_VALUE_EXCEL: cls._configure_excel,
+            cls._STORAGE_TYPE_VALUE_IN_MEMORY: cls._configure_in_memory,
+            cls._STORAGE_TYPE_VALUE_GENERIC: cls._configure_generic,
+            cls._STORAGE_TYPE_VALUE_JSON: cls._configure_json,
+            cls._STORAGE_TYPE_VALUE_PARQUET: cls._configure_parquet,
+        }
+
+        if storage_type in cls._ALL_STORAGE_TYPES:
+            return configuration_map[storage_type](id=id, scope=scope, **properties)
+
+        return cls.__configure(id, storage_type, scope, **properties)
 
-    @staticmethod
+    @classmethod
```

Comment:
This is updated to `@classmethod` since I need to access to the `cls._OPTIONAL_...` everywhere.